src/Dafda/Consuming/MessageFilters/MultiHeaderMessageFilter.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dafda.Configuration;
+using Dafda.Serializing;
+
+namespace Dafda.Consuming.MessageFilters
+{
+    /// <inheritdoc />
+    public class MultiHeaderMessageFilter : MessageFilter
```

Comment:
It might be worth providing logging versions of these OOTB? (but ofc not required for this PR)

Reply:
Maaaaaaaaybe, but will sort out some time I think. I wanted to involve more logging stuff here. Would be a task in it's own I think.